Common/src/containers/CFileReaderLUT.cpp

```diff
@@ -128,13 +172,19 @@ void CFileReaderLUT::ReadRawLUT(const string& file_name) {
 
   /*--- now that n_variables, n_points, n_hull_points and n_variables are available, allocate memory ---*/
   if (rank == MASTER_NODE) cout << "allocating memory for the data, size = ( " << GetNVariables() << " , " << GetNPoints() << " )" << endl;
-  table_data.resize(GetNVariables(), GetNPoints());
+  table_data = new su2activematrix[n_levels];
+  for(unsigned long i_level=0; i_level<n_levels; i_level++)
+    table_data[i_level].resize(GetNVariables(), GetNPoints(i_level));
 
   if (rank == MASTER_NODE) cout << "allocating memory for the triangles, size = "<< GetNTriangles() << endl;
-  triangles.resize(GetNTriangles(), 3);
+  triangles = new su2matrix<unsigned long>[n_levels];
+  for(unsigned long i_level=0; i_level<n_levels; i_level++)
+    triangles[i_level].resize(GetNTriangles(i_level), 3);
 
   if (rank == MASTER_NODE) cout << "allocating memory for the hull points, size = " << GetNHullPoints() << endl;
-  hull.resize(GetNHullPoints());
+  hull = new su2vector<unsigned long>[n_levels];
+  for(unsigned long i_level=0; i_level<n_levels; i_level++)
+    hull[i_level].resize(GetNHullPoints(i_level));
```

Comment:
If these cannot be a matrix then use std::vector<std::vector<>>, just don't use new basically.

Reply:
I see, thanks! Is this more memory efficient than the array? Or just a way to save some potential trouble with destructors?